staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -184,19 +184,19 @@ func (e *eventBroadcasterImpl) recordToSink(event *eventsv1.Event, clock clock.C
 					Count:            1,
 					LastObservedTime: metav1.MicroTime{Time: clock.Now()},
 				}
-				return isomorphicEvent
+				// Make a copy to be able to continue updating the cache in parallel.
```

Comment:
I think the comment is slightly misleading.
The isomorphicEvent is already stored in cache. So we're making a deep-copy actually to ensure that the recording of the event will not mess up with the object stored in the cache, right?

Same below.

Reply:
That is correct, I updated my comment to make that clearer :+1: